src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -520,19 +529,37 @@ private void ProjectFinished(object sender, ProjectFinishedEventArgs e)
                                 urlString = uri.AbsoluteUri;
                             }
 
+                            var relativeDisplayPathSpan = outputPathSpan;
+                            var workingDirectorySpan = _initialWorkingDirectory.AsSpan();
                             // If the output path is under the initial working directory, make the console output relative to that to save space.
-                            if (outputPathSpan.StartsWith(_initialWorkingDirectory.AsSpan(), FileUtilities.PathComparison))
+                            if (outputPathSpan.StartsWith(workingDirectorySpan, FileUtilities.PathComparison))
+                            {
+                                if (outputPathSpan.Length > workingDirectorySpan.Length
+                                    && (outputPathSpan[workingDirectorySpan.Length] == Path.DirectorySeparatorChar
+                                        || outputPathSpan[workingDirectorySpan.Length] == Path.AltDirectorySeparatorChar))
```

Comment:
I know this is pre-existing logic, but it feels fragile - what if in future the working directory path ends with separator?

It would be nice to unify and extract the logic of the 2 cases - "given the common root, CWD and a path - return a proper relative path".
As a side benefit, the +/-1 problem of the separator location would get easily testable.

Reply:
Yeah, I agree it's messy :) I can take a look and try to make something principled/reusable.

Reply:
I'm not at all sure that we need to care about Spans here - this logic will be run once per project, not every message.

Reply:
I rewrote this completely in terms of `Path.GetRelativePath`. To me this reads much more simply now - what do you think?

Reply:
Yeah - it now reads much easier - thanks for the effort!